eva-lib/src/main/java/uk/ac/ebi/variation/eva/lib/datastore/DBAdaptorConnector.java

```diff
@@ -105,7 +104,11 @@ public static MongoClient getMongoClient(Properties properties) throws UnknownHo
                     properties.getProperty("eva.mongo.passwd").toCharArray()));
         }
 
-        return new MongoClient(servers, mongoCredentialList);
+        MongoClientOptions options = MongoClientOptions.builder()
+                .readPreference(ReadPreference.valueOf(properties.getProperty("eva.mongo.read-preference")))
```

Comment:
A reasonable default for this `getProperty` would be "secondaryPreferred". It follows systems' recommendation, and if there are no secondaries then it automatically sends the requests to the primary.